src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -819,6 +819,14 @@ export class RoslynLanguageServer {
         args.push('--devKitDependencyPath', devKitDepsPath);
 
         args.push('--sessionId', getSessionId());
+
+        // Also include the Xaml Dev Kit extensions
+        const xamlBasePath = path.join(clientRoot, '..', '.xamlDesignTools', 'lib', 'netstandard2.0');
```

Comment:
TODO - Allow local overrides for these extension paths (to allow locally built binaries to replace the ones shipped in the extension).

Reply:
TODO - Unify path constants between package download and reading them in the extension.

Reply:
@dibarbet - can you say how you're thinking local overrides for the extension paths, for locally built binary binaries, would work - especially if you have any existing conventions there

Reply:
@BretJohnson I don't yet - the only overrides we have are for the roslyn and razor serverdlls directly (e.g. `dotnet.server.path`).  But that currently doesn't allow you to override the roslynDevkit path for instnace.
Some options I'm thinking of
1.  Look too see if the same dll name was specified in the extension paths setting, and using that path instead if set.
2.  Have some kind of `dotnet.server.componentOverrides` setting, perhaps like
```
dotnet.server.componentOverrides: {
    "roslynDevkit": <path to roslyndevkit drop>
    "xamlDesignTools": <path to xaml design tools drop>
}
```

It would be definitely configured in a vscode setting (similar to dotnet.server.path) though.

Reply:
this is what I ended up doing
```
"dotnet.server.componentPaths": {
    "roslynDevKit": "",
    "xamlDesignTools": "C:\\Users\\dabarbet\\Downloads\\.xamlDesignTools"
},
```

![image](https://github.com/dotnet/vscode-csharp/assets/5749229/8a33a6ef-d993-4764-bfbc-f55ea0149e7a)

Reply:
Probably this should get renamed to `xamlTools` too. (And thanks for sharing this, so I can copy some of the code.)